detectable/src/main/java/com/synopsys/integration/detectable/detectables/gradle/inspection/parse/GradleReportLineParser.java

```diff
@@ -77,9 +92,53 @@ private List<String> parseGav(String line) {
             }
         }
 
+        String projectName = metadata.getOrDefault(PROJECT_NAME_PREFIX, "orphanProject");
+        String rootProjectName = metadata.getOrDefault(ROOT_PROJECT_NAME_PREFIX, "");
+        String projectParent = metadata.getOrDefault(PROJECT_PARENT_PREFIX, null);
+
+        addRelation(projectParent, projectName, rootProjectName);
+
+        if(gavPieces.size() == 3) {
+            String dependencyGroupName = gavPieces.get(0) + ":" + gavPieces.get(1);
+            if ((cleanedOutput.contains(STRICTLY) || cleanedOutput.contains(REJECT) || cleanedOutput.contains(REQUIRE) || cleanedOutput.contains(PREFER))) {
+                gradleRichVersions.putIfAbsent(dependencyGroupName, gavPieces.get(2));
+                richVersionGroup.computeIfAbsent(projectName, value -> new HashSet<>()).add(dependencyGroupName);
+            }
+
+            if(gradleRichVersions.containsKey(dependencyGroupName)) {
+                if(richVersionGroup.containsKey(projectName) && richVersionGroup.get(projectName).contains(dependencyGroupName)) {
+                    gavPieces.set(2, gradleRichVersions.get(dependencyGroupName));
+                } else if(richVersionGroup.containsKey(rootProjectName) && richVersionGroup.get(rootProjectName).contains(dependencyGroupName)) {
+                    gavPieces.set(2, gradleRichVersions.get(dependencyGroupName));
+                } else if (checkParentRichVersion(rootProjectName, projectParent, dependencyGroupName)) {
+                    gavPieces.set(2, gradleRichVersions.get(dependencyGroupName));
+                }
```

Comment:
There are three cases where we have to apply the rich versions:
1. If the rich version dependency is declared in the current project.
2. If the rich version dependency is declared in the root project.
3. If the rich version dependency is declared in one of the many parents that this project could have moving up the hierarchy.